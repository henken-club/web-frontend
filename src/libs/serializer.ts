export type ContentBook<T extends {id: string}> = {
  type: 'Book';
  book: T;
};

export type ContentBookSeries<T extends {id: string}> = {
  type: 'BookSeries';
  bookSeries: T;
};

export type SerializeContent<
  TInput,
  TBook extends {id: string},
  TBookSeries extends {id: string},
> = (input: TInput) => ContentBook<TBook> | ContentBookSeries<TBookSeries>;

export const serializeUser = <T extends Record<string, unknown>>({
  __typename,
  ...props
}: {__typename: 'User'} & T) => ({...props});

export const serializeHenken = <T extends Record<string, unknown>>({
  __typename,
  ...props
}: {__typename: 'Henken'} & T) => ({...props});
