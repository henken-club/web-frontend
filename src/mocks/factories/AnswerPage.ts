import {
  AllAnswerPagesQuery,
  AllAnswerPagesQueryVariables,
  AnswerPageQuery,
  AnswerPageQueryVariables,
} from '../codegen';

import {
  alias,
  answerType,
  avatar,
  bookCover,
  comment,
  createdAt,
  displayName,
  id,
  repeat,
  title,
  which,
} from './common';

export const factoryAllAnswerPages = ({
  limit,
}: AllAnswerPagesQueryVariables): AllAnswerPagesQuery => ({
  __typename: 'Query',
  manyAnswers: repeat(limit, () => ({
    __typename: 'Answer',
    id: id(),
  })),
});

export const factoryAnswerPage = (
  variables: AnswerPageQueryVariables,
): AnswerPageQuery => ({
  __typename: 'Query',
  findAnswer: {
    __typename: 'FindAnswerPayload',
    answer: {
      __typename: 'Answer',
      id: id(),
      comment: comment(),
      createdAt: createdAt(),
      type: answerType(),
      answerTo: {
        __typename: 'Henken',
        id: id(),
        comment: comment(),
        postsTo: {
          __typename: 'User',
          id: id(),
          alias: alias(),
          displayName: displayName(),
          avatar: avatar(),
        },
        postedBy: {
          __typename: 'User',
          id: id(),
          alias: alias(),
          displayName: displayName(),
          avatar: avatar(),
        },
        content: which([
          {__typename: 'Book', id: id(), title: title(), cover: bookCover()},
          {__typename: 'BookSeries', id: id(), title: title()},
        ]),
      },
    },
  },
});
