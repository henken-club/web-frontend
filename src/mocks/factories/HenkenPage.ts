import {
  AllHenkenPagesQuery,
  AllHenkenPagesQueryVariables,
  HenkenPageQuery,
  HenkenPageQueryVariables,
} from '../codegen';

import {
  alias,
  authorName,
  avatar,
  bookCover,
  comment,
  displayName,
  id,
  repeat,
  title,
  which,
} from './common';

export const factoryAllHenkenPages = ({
  limit,
}: AllHenkenPagesQueryVariables): AllHenkenPagesQuery => ({
  __typename: 'Query',
  manyHenkens: {
    __typename: 'HenkenConnection',
    edges: repeat(limit, () => ({
      __typename: 'HenkenEdge',
      node: {
        __typename: 'Henken',
        id: id(),
      },
    })),
  },
});

export const factoryHenkenPage = (
  variables: HenkenPageQueryVariables,
): HenkenPageQuery => ({
  __typename: 'Query',
  findHenken: {
    __typename: 'FindHenkenPayload',
    henken: {
      __typename: 'Henken',
      id: variables.id,
      comment: comment(),
      postsTo: {
        __typename: 'User',
        id: id(),
        alias: alias(),
        displayName: displayName(),
        avatar: avatar(),
      },
      postedBy: {
        __typename: 'User',
        id: id(),
        alias: alias(),
        displayName: displayName(),
        avatar: avatar(),
      },
      content: which([
        {
          __typename: 'Book',
          id: id(),
          title: title(),
          cover: bookCover(),
        },
        {
          __typename: 'BookSeries',
          id: id(),
          title: title(),
        },
        {
          __typename: 'Author',
          id: id(),
          name: authorName(),
        },
      ]),
    },
  },
});
