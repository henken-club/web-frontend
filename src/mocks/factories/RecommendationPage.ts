import {
  AllRecommendationsPagesQuery,
  AllRecommendationsPagesQueryVariables,
  RecommendationPageQuery,
  RecommendationPageQueryVariables,
} from '../codegen';

import {
  alias,
  avatar,
  bookCover,
  displayName,
  score,
  id,
  title,
  updatedAt,
  which,
  repeat,
  authorName,
} from './common';

export const factoryAllRecommendationsPages = ({
  limit,
}: AllRecommendationsPagesQueryVariables): AllRecommendationsPagesQuery => ({
  __typename: 'Query',
  manyRecommendations: {
    __typename: 'RecommendationConnection',
    edges: repeat(limit, () => ({
      __typename: 'RecommendationEdge',
      node: {
        __typename: 'Recommendation',
        id: id(),
      },
    })),
  },
});

export const factoryRecommendationPage = (
  variables: RecommendationPageQueryVariables,
): RecommendationPageQuery => ({
  __typename: 'Query',
  findRecommendation: {
    __typename: 'FindRecommendationPayload',
    recommendation: {
      __typename: 'Recommendation',
      id: variables.id,
      score: score(),
      updatedAt: updatedAt(),
      recommendsTo: {
        __typename: 'User',
        id: id(),
        alias: alias(),
        displayName: displayName(),
        avatar: avatar(),
      },
      content: which([
        {__typename: 'Book', id: id(), title: title(), cover: bookCover()},
        {__typename: 'BookSeries', id: id(), title: title()},
        {__typename: 'Author', id: id(), name: authorName()},
      ]),
    },
  },
});
