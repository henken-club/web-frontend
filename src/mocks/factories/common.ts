import faker from 'faker';

import {AnswerType} from '../codegen';

// common
export const id = faker.datatype.uuid;
export const createdAt = () =>
  faker.date.between('2020-01-01', '2020-12-31').toISOString();
export const updatedAt = createdAt;

export const totalCount = faker.datatype.number;
export const hasNextPage = faker.datatype.boolean;

// Util
export const which = faker.random.arrayElement;
export const repeat = <T>(count: number, generator: () => T): T[] =>
  [...new Array(count)].map((_, i) => generator());
export const between = (min: number, max: number) =>
  min + faker.datatype.number(max - min);

// User
export const alias = faker.lorem.word;
export const displayName = faker.name.firstName;
export const avatar = () =>
  which([
    '/.mock/avatar_1.png',
    '/.mock/avatar_2.png',
    '/.mock/avatar_3.png',
    '/.mock/avatar_4.png',
    '/.mock/avatar_5.png',
    '/.mock/avatar_6.png',
    '/.mock/avatar_7.png',
    '/.mock/avatar_8.png',
    '/.mock/avatar_9.png',
    '/.mock/avatar_10.png',
    '/.mock/avatar_11.png',
    '/.mock/avatar_12.png',
  ]);

// Henken, Answer
export const title = faker.lorem.words;
export const comment = faker.lorem.words;

// Answer
export const answerType = () => which([AnswerType.Right, AnswerType.Wrong]);

// Recommendation
export const score = faker.datatype.number;

// Book
export const bookCover = () =>
  which([
    '/.mock/bookcover_1.jpg',
    '/.mock/bookcover_2.jpg',
    '/.mock/bookcover_3.jpg',
    '/.mock/bookcover_4.jpg',
    '/.mock/bookcover_5.jpg',
    '/.mock/bookcover_6.jpg',
    '/.mock/bookcover_7.jpg',
    '/.mock/bookcover_8.jpg',
  ]);

// Author
export const authorName = () => faker.name.findName();
