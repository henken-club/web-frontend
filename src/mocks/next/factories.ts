import faker from 'faker';

export const factoryUser = ({
  ...props
}: {
  id?: string;
  alias?: string;
  displayName?: string;
  avatar?: string;
} = {}) => ({
  __typename: 'User' as const,
  id: faker.datatype.uuid(),
  alias: faker.lorem.word(),
  displayName: faker.name.firstName(),
  avatar: faker.image.avatar(),
  ...props,
});

export const factoryUserEdge = ({
  node = [],
}: {node?: Parameters<typeof factoryUser>} = {}) => ({
  __typename: 'UserEdge' as const,
  node: factoryUser(...node),
});
