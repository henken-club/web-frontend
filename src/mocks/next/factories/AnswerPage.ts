import faker from 'faker';

import {
  AllAnswerPagesQuery,
  AllAnswerPagesQueryVariables,
  AnswerPageQuery,
} from '../codegen';

export const factoryAllAnswerPagesQuery = ({
  limit,
}: AllAnswerPagesQueryVariables): AllAnswerPagesQuery => ({
  __typename: 'Query',
  manyAnswers: [...new Array(limit)].map((_, i) => ({
    __typename: 'Answer',
    id: faker.datatype.uuid(),
  })),
});

export const factoryAnswerPageQuery = (): AnswerPageQuery => ({
  __typename: 'Query',
  findAnswer: {
    __typename: 'FindAnswerPayload',
    answer: {
      __typename: 'Answer',
      id: faker.datatype.uuid(),
    },
  },
});
