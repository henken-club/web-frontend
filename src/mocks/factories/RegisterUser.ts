import {RegisterUserMutation, RegisterUserMutationVariables} from '../codegen';

import {id} from './common';

export function factoryRegisterUser(
  variables: RegisterUserMutationVariables,
): RegisterUserMutation {
  return {
    __typename: 'Mutation',
    registerUser: {
      __typename: 'RegisterUserPayload',
      user: {__typename: 'User', id: id(), ...variables},
    },
  };
}
