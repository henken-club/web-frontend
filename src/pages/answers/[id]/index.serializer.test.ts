import faker from 'faker';

import {serializer} from './index.serializer';

import {factoryAnswerPage} from '~/mocks/factories/AnswerPage';

describe('AnswerPage serializer', () => {
  describe('serializer()', () => {
    it('findAnswer.answerがnullならnullを返却', () => {
      const actual = serializer({
        __typename: 'Query',
        findAnswer: {__typename: 'FindAnswerPayload', answer: null},
      });
      expect(actual).toBeNull();
    });

    it.each(
      [...new Array(10)].map((_, i) => {
        faker.seed(i);
        return [factoryAnswerPage()];
      }),
    )('findAnswer.answerが存在する場合 %#', (payload) => {
      const actual = serializer(payload);
      expect(actual).toMatchSnapshot();
    });
  });
});
