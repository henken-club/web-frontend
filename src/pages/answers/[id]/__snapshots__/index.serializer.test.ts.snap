// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 0 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "commodi quia similique",
      "content": Object {
        "bookSeries": Object {
          "id": "b4f63c96-9e9a-43ef-a77d-fb14cd66ae39",
          "title": "enim aut delectus",
        },
        "type": "BookSeries",
      },
      "id": "20592939-6fea-4759-aeb1-0faaa2352c59",
      "postedBy": Object {
        "alias": "eligendi",
        "avatar": "https://cdn.fakercloud.com/avatars/lmjabreu_128.jpg",
        "displayName": "Kenna",
        "id": "9da1ffe7-8f09-47b0-874f-15471b5e6e13",
      },
      "postsTo": Object {
        "alias": "qui",
        "avatar": "https://cdn.fakercloud.com/avatars/andrewcohen_128.jpg",
        "displayName": "Adrain",
        "id": "07aff1a3-a2fa-4946-b739-251aa52c3f5a",
      },
    },
    "comment": "est id blanditiis",
    "createdAt": "2020-03-20T00:58:52.049Z",
    "id": "a05dfc2d-df7c-4c78-8a1b-a928fc816742",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 1 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "magni cupiditate sit",
      "content": Object {
        "book": Object {
          "cover": null,
          "id": "5ec6ef0a-7911-4cf1-b59f-663cec940c92",
          "title": "voluptas culpa non",
        },
        "type": "Book",
      },
      "id": "3b452acd-600c-49fa-b447-c31177e14e41",
      "postedBy": Object {
        "alias": "ut",
        "avatar": "https://cdn.fakercloud.com/avatars/derekebradley_128.jpg",
        "displayName": "Norberto",
        "id": "cf09ee82-72ec-4668-aae5-59cab2eb9ac5",
      },
      "postsTo": Object {
        "alias": "nam",
        "avatar": "https://cdn.fakercloud.com/avatars/kerihenare_128.jpg",
        "displayName": "Brook",
        "id": "1ae3e447-9099-492f-94b3-186fb7680c80",
      },
    },
    "comment": "quam tempore a",
    "createdAt": "2020-05-30T07:15:42.551Z",
    "id": "e67862f3-ccbf-4c51-b8ed-ed1d0420ea19",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 2 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "tenetur quibusdam perspiciatis",
      "content": Object {
        "book": Object {
          "cover": null,
          "id": "e6d568af-5165-4f0a-bfac-336ea9ac28ef",
          "title": "omnis in dolores",
        },
        "type": "Book",
      },
      "id": "c8a69cf9-122b-4df8-83ef-55946c65c784",
      "postedBy": Object {
        "alias": "iure",
        "avatar": "https://cdn.fakercloud.com/avatars/justinrgraham_128.jpg",
        "displayName": "Oceane",
        "id": "efb788e5-8496-46de-b06b-05c3459e05a8",
      },
      "postsTo": Object {
        "alias": "ex",
        "avatar": "https://cdn.fakercloud.com/avatars/adamawesomeface_128.jpg",
        "displayName": "Nolan",
        "id": "d1f5dd86-30b3-461e-bff8-cb9bc72f4685",
      },
    },
    "comment": "sed totam rerum",
    "createdAt": "2020-08-26T19:22:18.319Z",
    "id": "4d876d21-f841-4461-9227-99331c395275",
    "type": "right",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 3 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "blanditiis architecto quia",
      "content": Object {
        "book": Object {
          "cover": null,
          "id": "585b846e-dd9b-4e7c-95bc-aeeca45e883d",
          "title": "consectetur ut est",
        },
        "type": "Book",
      },
      "id": "66616ac8-b503-4569-bb49-437ada9d4c45",
      "postedBy": Object {
        "alias": "tenetur",
        "avatar": "https://cdn.fakercloud.com/avatars/nathalie_fs_128.jpg",
        "displayName": "Malvina",
        "id": "cf530b7a-0b87-4e93-9e14-322f1b2c8131",
      },
      "postsTo": Object {
        "alias": "veritatis",
        "avatar": "https://cdn.fakercloud.com/avatars/d_nny_m_cher_128.jpg",
        "displayName": "Eliza",
        "id": "c318d1b7-755b-48b7-bcb5-5da7589d5872",
      },
    },
    "comment": "tempore voluptas ut",
    "createdAt": "2020-10-22T06:13:11.238Z",
    "id": "e0c8f466-844b-4974-a78c-cd443e6be0f2",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 4 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "fuga quae saepe",
      "content": Object {
        "bookSeries": Object {
          "id": "61e3f09e-1eb8-4abd-b500-79214cccc6ef",
          "title": "non quia sunt",
        },
        "type": "BookSeries",
      },
      "id": "0238382a-563a-466a-93c2-296587a54184",
      "postedBy": Object {
        "alias": "sint",
        "avatar": "https://cdn.fakercloud.com/avatars/looneydoodle_128.jpg",
        "displayName": "Tobin",
        "id": "19031b3e-c819-411f-8229-9f55c00814ad",
      },
      "postsTo": Object {
        "alias": "mollitia",
        "avatar": "https://cdn.fakercloud.com/avatars/aluisio_azevedo_128.jpg",
        "displayName": "Marcellus",
        "id": "e53ac327-12ae-4f65-9e6b-e56e569ddbdb",
      },
    },
    "comment": "itaque fugiat occaecati",
    "createdAt": "2020-04-05T06:06:55.480Z",
    "id": "c09640af-c60f-40cf-9727-11922dbd6e8d",
    "type": "right",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 5 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "unde est assumenda",
      "content": Object {
        "bookSeries": Object {
          "id": "0b3f79b8-5359-4ecf-963e-d521c8eabe2f",
          "title": "quia assumenda id",
        },
        "type": "BookSeries",
      },
      "id": "909950b8-daef-404b-8ad3-eb0979f162d5",
      "postedBy": Object {
        "alias": "ut",
        "avatar": "https://cdn.fakercloud.com/avatars/angelceballos_128.jpg",
        "displayName": "Fermin",
        "id": "983c20d3-d5a3-42c2-95af-739efef11cdf",
      },
      "postsTo": Object {
        "alias": "ea",
        "avatar": "https://cdn.fakercloud.com/avatars/_yardenoon_128.jpg",
        "displayName": "Henry",
        "id": "3710e141-740f-44a3-9f96-a6c4e3ebde1a",
      },
    },
    "comment": "sed nobis cumque",
    "createdAt": "2020-11-03T19:58:02.295Z",
    "id": "3614ea89-79b4-4041-8d5d-2e2ff8f13f07",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 6 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "assumenda vero voluptas",
      "content": Object {
        "bookSeries": Object {
          "id": "219d4ec3-21cd-49d0-880a-cfac9ce7a04d",
          "title": "qui perferendis asperiores",
        },
        "type": "BookSeries",
      },
      "id": "b9bbcee5-d446-4bb6-b76b-f57e5bb31a60",
      "postedBy": Object {
        "alias": "et",
        "avatar": "https://cdn.fakercloud.com/avatars/kennyadr_128.jpg",
        "displayName": "Jaydon",
        "id": "fd8a827b-6063-44e0-a629-a82aa14a0cf3",
      },
      "postsTo": Object {
        "alias": "ea",
        "avatar": "https://cdn.fakercloud.com/avatars/petebernardo_128.jpg",
        "displayName": "Tyson",
        "id": "748327c5-9b33-4658-9898-e7fade279674",
      },
    },
    "comment": "asperiores perspiciatis aliquam",
    "createdAt": "2020-10-14T11:27:41.339Z",
    "id": "ed76aecd-70ab-42c5-b9b7-8911f3633db7",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 7 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "quod ullam nihil",
      "content": Object {
        "book": Object {
          "cover": null,
          "id": "7817cc7e-3aec-44f3-85e6-4372c6bd2caa",
          "title": "quasi dolores ex",
        },
        "type": "Book",
      },
      "id": "554d7cf5-2924-4178-94a4-5e76b66fec25",
      "postedBy": Object {
        "alias": "est",
        "avatar": "https://cdn.fakercloud.com/avatars/rpatey_128.jpg",
        "displayName": "Sanford",
        "id": "7da7add2-8407-41bc-b62a-58564c2f6478",
      },
      "postsTo": Object {
        "alias": "delectus",
        "avatar": "https://cdn.fakercloud.com/avatars/uxalex_128.jpg",
        "displayName": "Bud",
        "id": "869aa846-60a1-47bc-8abb-f7af024a983d",
      },
    },
    "comment": "quos laborum quis",
    "createdAt": "2020-12-13T04:56:07.866Z",
    "id": "b357ee40-c94f-4d32-8cef-2786c5a37137",
    "type": "right",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 8 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "ducimus autem amet",
      "content": Object {
        "book": Object {
          "cover": null,
          "id": "399d889f-4b62-4896-84ce-c6432ea602c1",
          "title": "quas et repellat",
        },
        "type": "Book",
      },
      "id": "af452614-ac1a-471a-92ca-007ec3f99402",
      "postedBy": Object {
        "alias": "voluptatum",
        "avatar": "https://cdn.fakercloud.com/avatars/juangomezw_128.jpg",
        "displayName": "Rashawn",
        "id": "c2f819bb-f4b6-47d1-9cb0-81d3b5de3747",
      },
      "postsTo": Object {
        "alias": "iusto",
        "avatar": "https://cdn.fakercloud.com/avatars/j04ntoh_128.jpg",
        "displayName": "Donavon",
        "id": "7a43d2d8-b17b-48fa-b75b-97a98a770010",
      },
    },
    "comment": "aliquid ad vel",
    "createdAt": "2020-07-27T04:13:16.143Z",
    "id": "547fd503-414f-4a24-9217-386fe85af105",
    "type": "wrong",
  },
}
`;

exports[`AnswerPage serializer serializer() findAnswer.answerが存在する場合 9 1`] = `
Object {
  "answer": Object {
    "answerTo": Object {
      "comment": "eaque maiores est",
      "content": Object {
        "bookSeries": Object {
          "id": "e28f2e11-db87-44c9-b5cc-57ef880ee398",
          "title": "perferendis quas sunt",
        },
        "type": "BookSeries",
      },
      "id": "a39dce0c-9517-4296-b972-6edf2eb6b4e5",
      "postedBy": Object {
        "alias": "ut",
        "avatar": "https://cdn.fakercloud.com/avatars/colgruv_128.jpg",
        "displayName": "Amos",
        "id": "c50737b6-5383-403d-b868-9d7065b98242",
      },
      "postsTo": Object {
        "alias": "autem",
        "avatar": "https://cdn.fakercloud.com/avatars/ceekaytweet_128.jpg",
        "displayName": "Stewart",
        "id": "0a274060-a23c-4222-a15a-7db5c04ce701",
      },
    },
    "comment": "eos molestiae ut",
    "createdAt": "2020-12-23T22:08:54.317Z",
    "id": "2e2af88b-f6ab-49da-bcf6-cdf12181f7a3",
    "type": "right",
  },
}
`;
